server/src/internalClusterTest/java/org/opensearch/remotemigration/RemoteDualMigrationIT.java

```diff
@@ -0,0 +1,483 @@
+                .admin()
+                .cluster()
+                .prepareGetRepositories(REPOSITORY_NAME, REPOSITORY_2_NAME)
+                .get()
+                .repositories()
+                .size(),
+            2
+        );
+
+        int firstBatch = randomIntBetween(1, 100);
+        logger.info("---> Indexing {} docs", firstBatch);
+        indexBulk(REMOTE_PRI_DOCREP_REMOTE_REP, firstBatch);
+
+        String primaryShardHostingNode = primaryNodeName(REMOTE_PRI_DOCREP_REMOTE_REP);
+        logger.info("---> Moving primary copy from {} to remote enabled node {}", primaryShardHostingNode, remoteNodeName);
+        assertAcked(
+            internalCluster().client()
+                .admin()
+                .cluster()
+                .prepareReroute()
+                .add(new MoveAllocationCommand(REMOTE_PRI_DOCREP_REMOTE_REP, 0, primaryShardHostingNode, remoteNodeName))
+                .get()
+        );
+        ensureGreen(REMOTE_PRI_DOCREP_REMOTE_REP);
+        ClusterState clusterState = internalCluster().client().admin().cluster().prepareState().get().getState();
+        String primaryShardHostingNodeId = clusterState.getRoutingTable()
+            .index(REMOTE_PRI_DOCREP_REMOTE_REP)
+            .shard(0)
+            .primaryShard()
+            .currentNodeId();
+        assertTrue(clusterState.getNodes().get(primaryShardHostingNodeId).isRemoteStoreNode());
+
+        logger.info("---> Starting another remote enabled node");
+        internalCluster().startDataOnlyNode();
+        internalCluster().validateClusterFormed();
+
+        logger.info("---> Expanding index to 2 replica copies");
+        Settings twoReplicas = Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 2).build();
+        assertAcked(
+            internalCluster().client()
+                .admin()
+                .indices()
+                .prepareUpdateSettings()
+                .setIndices(REMOTE_PRI_DOCREP_REMOTE_REP)
+                .setSettings(twoReplicas)
+                .get()
+        );
+        ensureGreen(REMOTE_PRI_DOCREP_REMOTE_REP);
+
+        int secondBatch = randomIntBetween(1, 10);
+        logger.info("---> Indexing another {} docs", secondBatch);
+        indexBulk(REMOTE_PRI_DOCREP_REMOTE_REP, secondBatch);
+        // Defensive check to ensure that doc count in replica shard catches up to the primary copy
+        refreshAndWaitForReplication(REMOTE_PRI_DOCREP_REMOTE_REP);
+        assertReplicaAndPrimaryConsistency(REMOTE_PRI_DOCREP_REMOTE_REP, firstBatch, secondBatch);
+    }
+
+    /*
+    Checks if retention leases are published on primary shard and it's docrep copies, but not on remote copies
+     */
+    public void testRetentionLeasePresentOnDocrepReplicaButNotRemote() throws Exception {
+        testRemotePrimaryDocRepAndRemoteReplica();
+        // Ensuring global checkpoint is synced across all the shard copies
+        persistGlobalCheckpoint(REMOTE_PRI_DOCREP_REMOTE_REP);
```

Comment:
Calling this should not be required since the bulk indexing is expected to call the sync method.

Reply:
Removed in latest rev